docs/testing-teal.md

```diff
@@ -0,0 +1,80 @@
+# Testing TEAL
+
+[TEAL](https://developer.algorand.org/docs/reference/teal/specification/) is a bytecode based stack language that executes inside Algorand transactions to check the parameters of the transaction and approve the transaction as if by a signature. `@algorand-builder/algorand-js` provides a simple workflow to execute TEAL code with high speed and throughput (as the transactions are not processed on the blockchain but in a test environment).
+
+**NOTE:** TEAL cannot modify or create transactions, only reject or approve them. Approval is signaled by finishing with the stack containing a single non-zero uint64 value.
+
+## How it works
+
+`@algorand-builder/algorand-js` (JavaScript Algorand runtime) executes transactions and processes TEAL in 3 parts :-
+
+- [Runtime](../packages/algorand-js/src/runtime): For a transaction or txn group, the state management is handled by the `Runtime`. User can use `Runtime` class to set up accounts, create applications, opt-in to app, update app..etc
+- [Parser](../packages/algorand-js/src/parser): Reads TEAL code and converts it to a list of opcodes which are executable by the interpreter. If any opcode/data in teal code is invalid, parser will throw an error.
+- [Interpreter](../packages/algorand-js/src/interpreter): Executes the list of opcodes returned by parser and updates stack after each execution. After the end of execution, if top of stack contains a non-zero uint64 value then the teal code is approved, and transaction can be executed.
+
```

Comment:
Please add also a documentation about `StoreAccountImpl`.

Reply:
Note, we should rename StoreAccountImpl: https://www.pivotaltracker.com/story/show/176449616

Reply:
ok